### PR TITLE
SCO-160: Steal the Wicket specific Sakai date-picker and spinner button from site-stats

### PR DESCRIPTION
**Pull request by bjones86**
https://jira.sakaiproject.org/browse/SCO-160

Statistics (specifically User Activity) has Wicket components for the standard Sakai date-picker and the spinner button. Let's copy these into the wicket-for-scorm internal library so we can use them in the SCORM Player. This will give the tool a more consistent look and feel with the rest of Sakai.

At a later date, we can rip the wicket-for-scorm internal library out of the SCORM Player project, and add it to core Sakai code as a general use Wicket library. At this point, things we copied from site-stats and gradebookng can be removed from their original projects in favour of the library's copy. This will reduce code duplication between all Wicket projects in Sakai.

### CODE REVIEW

**Comment by bjones86** [comment]
This PR depends on #34, #35, #36, #37, #38, and #39 which will need to be merged prior to this. Changes related to this ticket are isolated in c8a0202

File: scorm-tool/src/java/org/sakaiproject/scorm/ui/console/pages/PackageConfigurationPage.html [hunk 1]
```diff
@@ -1,97 +1,97 @@
...
+					<select wicket:id="numberOfTries"/>
+				</p>
+				<p class="shorttext">
+					<label for="hideTOC">
+						<wicket:message key="form.label.hide.toc"></wicket:message>
+					</label>
+					<input type="checkbox" wicket:id="hideTOC"></input>
+				</p>
+				<p class="shorttext">
+					<label for="createdBy">
```
Comment (plukasew) [inline scorm-tool/src/java/org/sakaiproject/scorm/ui/console/pages/PackageConfigurationPage.html]: Spans don't have labels

File: scorm-tool/src/java/org/sakaiproject/scorm/ui/console/pages/PackageRemovePage.java [hunk 2]
```diff
@@ -108,7 +108,7 @@ public FileRemoveForm( String id, final PageParameters params )
 			DataTable removeTable = new DataTable( "removeTable", columns, new ListDataProvider( list ), 3 );
 
 			final Label alertLabel = new Label( "alert", new ResourceModel( "verify.remove" ) );
-			final CancelButton btnCancel = new CancelButton( "btnCancel", PackageListPage.class );
+			final SakaiAjaxCancelButton btnCancel = new SakaiAjaxCancelButton( "btnCancel", PackageListPage.class );
```
Comment (plukasew) [inline scorm-tool/src/java/org/sakaiproject/scorm/ui/console/pages/PackageRemovePage.java:110]: There is an IndicatingAjaxButton btnSubmit that should also be a spinner button right? I also notice that these buttons disable each other, which is not something the normal SakaiAjaxButton can do (it just worries about itself). Perhaps we should enhance it to allow it to disable other page elements so we don't have to keep adding the extra AjaxCallListener?

File: scorm-tool/src/java/org/sakaiproject/scorm/ui/upload/UploadPage.java [hunk 3]
```diff
@@ -111,9 +111,9 @@ public FileUploadForm( String id )
             add( fileUploadField );
 
             // SCO-98 - disable buttons on submit, add spinner
-            final CancelButton btnCancel = new CancelButton( "btnCancel", PackageListPage.class );
+            final SakaiAjaxCancelButton btnCancel = new SakaiAjaxCancelButton( "btnCancel", PackageListPage.class );
             btnCancel.setOutputMarkupId( true );
-            IndicatingAjaxButton btnSubmit = new IndicatingAjaxButton( "btnSubmit", this )
+            SakaiAjaxButton btnSubmit = new SakaiAjaxButton( "btnSubmit", this )
```
Comment (plukasew) [inline scorm-tool/src/java/org/sakaiproject/scorm/ui/upload/UploadPage.java]: The part where it disables itself in updateAjaxAttributes is no longer needs, SakaiAjaxButton should already handle this.

**Review by plukasew — CHANGES_REQUESTED** [review]

File: scorm-tool/src/java/org/sakaiproject/scorm/ui/console/pages/PackageRemovePage.java (hunk 2 above)
Comment (bjones86) [reply scorm-tool/src/java/org/sakaiproject/scorm/ui/console/pages/PackageRemovePage.java:110]: Yes, good call on both. I've updated `SakaiAjaxButton` and its related `SakaiSpinnerAjaxCallListener` to take a list of element IDs which can be enabled/disabled on click.

**Pull request merged**